setuptools/package_index.py

```diff
@@ -720,20 +719,15 @@ def gen_setup(self, filename, fragment, tmpdir):
             with open(os.path.join(tmpdir, 'setup.py'), 'w', encoding="utf-8") as file:
                 file.write(
                     "from setuptools import setup\n"
-                    "setup(name=%r, version=%r, py_modules=[%r])\n"
-                    % (
-                        dists[0].project_name,
-                        dists[0].version,
-                        os.path.splitext(basename)[0],
-                    )
+                    f"setup(name={dists[0].project_name!r}, version={dists[0].version!r}, py_modules=[{os.path.splitext(basename)[0]!r}])\n"
```

Comment:
Another case where the line is so long, splitting or using `.format` is imo much more readable

Reply:
That , or perhaps use additional variables with meaningful names:
```python
                    name = dists[0].project_name
                    version = dists[0].version
                    modules = os.path.splitext(basename)[0]
                    f"setup(name={name!r}, version={version!r}, py_modules=[{modules!r}])\n"
```

Reply:
I have added the extra variables, and also changed `{name!r}` to `'{name}'`. I find the intent is clearer this way.

Reply:
You're right, both output the same result. Again, I suspect the author wanted to convey the very precise idea that `{str(self)!r}` printed into Python code will yield `str(self)`  upon execution. Both are equivalent in practice and your suggestion more concise, but I didn't feel like changing the code that much.

Reply:
I've reverted to `{str(self)!r}`, let's see whether Windows tests pass.